cluster/rbac/manager.go

```diff
@@ -114,6 +115,37 @@ func (m *Manager) UpsertRolesPermissions(c *cmd.ApplyRequest) error {
 		return fmt.Errorf("%w: %w", ErrBadRequest, err)
 	}
 
+	switch req.Version {
+	case cmd.RBACCommandPolicyVersionV0:
+		for roleName, policies := range req.Roles {
+			permissions := []*authorization.Policy{}
+			for _, p := range policies {
+				permissions = append(permissions, &p)
+			}
+			// remove old permissions
+			if err := m.authZ.RemovePermissions(roleName, permissions); err != nil {
+				return err
+			}
+
+			// create new permissions
+			for idx := range policies {
+				if req.Roles[roleName][idx].Domain != authorization.SchemaDomain {
+					continue
+				}
+
+				parts := strings.Split(req.Roles[roleName][idx].Resource, "/")
+				if len(parts) < 3 {
+					// shall never happens
+					return fmt.Errorf("invalid schema path")
+				}
+				req.Roles[roleName][idx].Resource = authorization.CollectionsMetadata(parts[2])[0]
+
+			}
+		}
```

Comment:
also making sure I understand it correctly:

- we remove all old permissions from the role just in case
- all permissions that need migrating to the new format will be changed in the role that are contained in the command
- the command is replayed and all permissions are added again

correct?

Reply:
correct, 

we make that because what we released as actions is actually now `create_tenants` and to make sure that if the users chose `create_collections` , shall stay `create_collections` by recreating it with new paths